test/framework/src/main/java/org/elasticsearch/AnalysisFactoryTestCase.java

```diff
@@ -0,0 +1,326 @@
+        put("trim",                      TrimTokenFilterFactory.class);
+        put("truncate",                  TruncateTokenFilterFactory.class);
+        put("turkishlowercase",          LowerCaseTokenFilterFactory.class);
+        put("type",                      KeepTypesFilterFactory.class);
+        put("uppercase",                 UpperCaseTokenFilterFactory.class);
+        put("worddelimiter",             WordDelimiterTokenFilterFactory.class);
+                
+        // TODO: these tokenfilters are not yet exposed: useful?
+
+        // suggest stop
+        put("suggeststop",               Void.class);
+        // capitalizes tokens
+        put("capitalization",            Void.class);
+        // like length filter (but codepoints)
+        put("codepointcount",            Void.class);
+        // puts hyphenated words back together
+        put("hyphenatedwords",           Void.class);
+        // repeats anything marked as keyword
+        put("keywordrepeat",             Void.class);
+        // like limittokencount, but by offset
+        put("limittokenoffset",          Void.class);
+        // like limittokencount, but by position
+        put("limittokenposition",        Void.class);
+        // ???
+        put("numericpayload",            Void.class);
+        // removes duplicates at the same position (this should be used by the existing factory)
+        put("removeduplicates",          Void.class);
+        // ???
+        put("tokenoffsetpayload",        Void.class);
+        // puts the type into the payload
+        put("typeaspayload",             Void.class);
+        // fingerprint
+        put("fingerprint",               Void.class);
+        // for tee-sinks
+        put("daterecognizer",            Void.class);
+    }});
+    
+    static final Map<String,Class<?>> KNOWN_CHARFILTERS = Collections.unmodifiableMap(new HashMap<String,Class<?>>() {{        
+        // exposed in ES
+        put("htmlstrip",      HtmlStripCharFilterFactory.class);
+        put("mapping",        MappingCharFilterFactory.class);
+        put("patternreplace", PatternReplaceCharFilterFactory.class);
+                
+        // TODO: these charfilters are not yet exposed: useful?
+        // handling of zwnj for persian
+        put("persian",        Void.class);
+    }});
+
+    protected Map<String, Class<?>> getTokenizers() {
+        return KNOWN_TOKENIZERS;
+    }
+
+    protected Map<String, Class<?>> getTokenFilters() {
+        return KNOWN_TOKENFILTERS;
+    }
+
+    protected Map<String, Class<?>> getCharFilters() {
+        return KNOWN_CHARFILTERS;
+    }
+
+    public void testTokenizers() {
+        Set<String> missing = new TreeSet<String>(org.apache.lucene.analysis.util.TokenizerFactory.availableTokenizers());
+        missing.removeAll(getTokenizers().keySet());
+        assertTrue("new tokenizers found, please update KNOWN_TOKENIZERS: " + missing.toString(), missing.isEmpty());
```

Comment:
Do you think it is worth switching it to `assertThat(missing, empty())` or something like that? I see this is managing the error message itself but I'm a fan of using hamcrest for it usually.